KubeArmor/enforcer/bpflsm/enforcer.go

```diff
@@ -394,10 +394,14 @@ func (be *BPFEnforcer) TraceEvents() {
 			log.DroppingAlertsInterval = cfg.GlobalCfg.ThrottleSec
 		}
 		// fallback logic if we don't receive source from BuildLogBase()
-		if log.Operation != "Process" && len(log.Source) == 0 {
+		if log.Operation == "Process" && len(log.Source) == 0 {
```

Comment:
Why is this required for `process`? We are already doing `log.Source = string(bytes.Trim(event.Data.Source[:], "\x00"))` at line380. This condition was implemented as a backup logic for other hooks since we were using source from `BuildLogBase`